examples/dataflow-streaming-benchmark/src/main/java/com/google/cloud/pso/pipeline/StreamingBenchmark.java

```diff
@@ -0,0 +1,205 @@
+
+    run(options);
+  }
+
+  /**
+   * Runs the pipeline to completion with the specified options. This method does not wait until the
+   * pipeline is finished before returning. Invoke {@code result.waitUntilFinish()} on the result
+   * object to block until the pipeline is finished running if blocking programmatic execution is
+   * required.
+   *
+   * @param options The execution options.
+   * @return The pipeline result.
+   */
+  public static PipelineResult run(Options options) {
+
+    // Create the pipeline
+    Pipeline pipeline = Pipeline.create(options);
+
+    /*
+     * Steps:
+     *  1) Trigger at the supplied QPS
+     *  2) Generate messages containing fake data
+     *  3) Write messages to Pub/Sub
+     */
+    pipeline
+        .apply(
+            "Trigger",
+            GenerateSequence.from(0L).withRate(options.getQps(), Duration.standardSeconds(1L)))
+        .apply("GenerateMessages", ParDo.of(new MessageGeneratorFn(options.getSchemaLocation())))
+        .apply("WriteToPubsub", PubsubIO.writeMessages().to(options.getTopic()));
+
+    return pipeline.run();
+  }
+
+  /**
+   * The {@link MessageGeneratorFn} class generates {@link PubsubMessage} objects from a supplied
+   * schema and populating the message with fake data.
+   *
+   * <p>See <a href="https://github.com/vincentrussell/json-data-generator">json-data-generator</a>
+   * for instructions on how to construct the schema file.
+   */
+  static class MessageGeneratorFn extends DoFn<Long, PubsubMessage> {
+
+    private final String schemaLocation;
+    private String schema;
+
+    MessageGeneratorFn(String schemaLocation) {
+      this.schemaLocation = schemaLocation;
+    }
+
+    @Setup
+    public void setup() throws IOException {
+      Metadata metadata = FileSystems.matchSingleFileSpec(schemaLocation);
+
+      // Copy the schema file into a string which can be used for generation.
+      try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+        try (ReadableByteChannel readerChannel = FileSystems.open(metadata.resourceId())) {
+          try (WritableByteChannel writerChannel = Channels.newChannel(byteArrayOutputStream)) {
+            ByteStreams.copy(readerChannel, writerChannel);
+          }
+        }
+
+        schema = byteArrayOutputStream.toString();
+      }
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext context)
+        throws IOException, JsonDataGeneratorException {
+
+      byte[] payload;
+      Map<String, String> attributes = Maps.newHashMap();
+
+      // Generate the fake JSON according to the schema.
+      try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+        JsonDataGeneratorImpl parser = new JsonDataGeneratorImpl();
+        parser.generateTestDataJson(schema, byteArrayOutputStream);
+
+        payload = byteArrayOutputStream.toByteArray();
+      }
+
+      context.output(new PubsubMessage(payload, attributes));
```

Comment:
Would it make sense to just publish json strings instead of a PubsubMessage object? We are not really passing any attributes so it might be easier for the downstream system to just consume UTF-8 encoded strings vs consuming PubsubMessage object.

Reply:
Eventually I imagine we can add functionality whereby the user can specify the eventId and eventTimestamp fields and we can populate those in the attributes. Added a TODO.